language-service/src/services/yamlCompletion.ts

```diff
@@ -98,518 +98,523 @@ export class YAMLCompletion {
+            case 'property': return CompletionItemKind.Property;
+            default: return CompletionItemKind.Value;
+        }
+    }
+
+    private getCurrentWord(document: TextDocument, offset: number) {
+        var i = offset - 1;
+        var text = document.getText();
+        while (i >= 0 && ' \t\n\r\v":{[,]}'.indexOf(text.charAt(i)) === -1) {
+            i--;
+        }
+        return text.substring(i + 1, offset);
+    }
+
+    private findItemAtOffset(node: Parser.ASTNode, document: TextDocument, offset: number) {
+        let scanner = Json.createScanner(document.getText(), true);
+        let children = node.getChildNodes();
+        for (let i = children.length - 1; i >= 0; i--) {
+            let child = children[i];
+            if (offset > child.end) {
+                scanner.setPosition(child.end);
+                let token = scanner.scan();
+                if (token === Json.SyntaxKind.CommaToken && offset >= scanner.getTokenOffset() + scanner.getTokenLength()) {
+                    return i + 1;
+                }
+                return i;
+            } else if (offset >= child.start) {
+                return i;
+            }
+        }
+        return 0;
+    }
+
+    private isInComment(document: TextDocument, start: number, offset: number) {
+        let scanner = Json.createScanner(document.getText(), false);
+        scanner.setPosition(start);
+        let token = scanner.scan();
+        while (token !== Json.SyntaxKind.EOF && (scanner.getTokenOffset() + scanner.getTokenLength() < offset)) {
+            token = scanner.scan();
+        }
+        return (token === Json.SyntaxKind.LineCommentTrivia || token === Json.SyntaxKind.BlockCommentTrivia) && scanner.getTokenOffset() <= offset;
+    }
+
+    private getInsertTextForPlainText(text: string): string {
+        return text.replace(/[\\\$\}]/g, '\\$&');   // escape $, \ and } 
+    }
+
+    private getInsertTextForValue(value: any, separatorAfter: string): string {
+        var text = value;
+        if (text === '{}') {
+            return '{\n\t$1\n}' + separatorAfter;
+        } else if (text === '[]') {
+            return '[\n\t$1\n]' + separatorAfter;
+        }
+        return this.getInsertTextForPlainText(text + separatorAfter);
+    }
+
+    private getInsertTextForProperty(key: string, propertySchema: JSONSchema, addValue: boolean, separatorAfter: string): string {
+
+        const propertyText = this.getInsertTextForValue(key, '');
+        const resultText = propertyText + ':';
+
+        let value: string = null;
+        if (propertySchema) {
+            var type = Array.isArray(propertySchema.type) ? propertySchema.type[0] : propertySchema.type;
+            if (!type) {
+                if (propertySchema.properties) {
+                    type = 'object';
+                } else if (propertySchema.items) {
+                    type = 'array';
+                } else if (propertySchema.hasOwnProperty("oneOf")) {
```

Comment:
Do you have specific cases for these?  Treating them as "object" might be a better match.

Reply:
Yeah, here is a specific example:
```
        "pool": {
            "oneOf": [
              {
                "type": "string"
              },
              {
                "$ref": "#/definitions/pool"
              }
            ],
            "description": "Pool where this job will run"
          },
```
When it is one of we do not know what it can really be. It can be object or it can be just string. In case of object we do put cursor on the next line, but in case of string we shouldn't. So I think it is better to leave the cursor on current line.